ckanext/xroad_integration/logic/action.py

```diff
@@ -485,16 +485,17 @@ def fetch_xroad_errors(context, data_dict):
                     error_log_list = error_data.get('errorLogList', [])
 
                     for error in error_log_list:
+                        service_version = error.get('serviceVersion', '')
                         mapped_error = {
                             "message": error['message'],
                             "code": error['code'],
                             "created": parse_xroad_catalog_datetime(error['created']),
                             "xroad_instance": error['xroadInstance'],
                             "member_class": error['memberClass'],
                             "member_code": error['memberCode'],
-                            "subsystem_code": error['subsystemCode'],
+                            "subsystem_code": error.get('subsystemCode', ''),
                             "service_code": error.get('serviceCode', ''),
-                            "service_version": error.get('serviceVersion', ''),
+                            "service_version": service_version if service_version is not None else '',
```

Comment:
service_version will never be None as you added default value in `error.get('serviceVersion', '')`, it will always be either the value from error dict or empty string